### PR TITLE
feat: add TUI based talos interactive installer

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/talos/issues/2742

This is initial commit of the installer.
What's done:
- verifying node availability before starting any operations.
- gathering information about disks on the machine.
- allows setting: install disk, hostname, machine type, installer image,
  kubernetes version, dns domain, cluster name.
- can dump/merge talosconfig to a file after applying configuration.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

**Comment by Unix4ever** [comment]
Unreachable:
[asciicast](https://asciinema.org/a/mQDmIafkDabbFjVvqo7p6mYBE)

Normal flow:
[asciicast](https://asciinema.org/a/pQSRCeIAfzf0L1GqsJs4VLlUR)

**Comment by Unix4ever** [comment]
Final form:
https://asciinema.org/a/5A2EjELKcgLH4NSgozhakIVqI

**Comment by andrewrynhard** [comment]
This looks awesome.

File: internal/integration/api/generate-config.go [hunk 1]
```diff
@@ -113,3 +113,7 @@ func (suite *GenerateConfigSuite) TestGenerate() {
 func init() {
 	allSuites = append(allSuites, new(GenerateConfigSuite))
 }
+
+func init() {
```
Comment (smira) [inline internal/integration/api/generate-config.go:117]: duplicate `func init()`

Comment (Unix4ever) [reply internal/integration/api/generate-config.go:117]: oh, that's merge artifact

**Review by smira — DISMISSED** [review]

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**